zerver/tests/test_slack_importer.py

```diff
@@ -2723,3 +2729,225 @@ def test_cancel_realm_import_realm_created(self) -> None:
                     "cancel_import": "true",
                 },
             )
+
+    @responses.activate
+    def test_slack_get_emojis(self) -> None:
+        zerver_realmemoji = [
+            {
+                "id": 0,
+                "name": "bowtie",
+                "file_name": None,
+                "is_animated": False,
+                "deactivated": False,
+                "author": None,
+                "realm": 0,
+            },
+            {
+                "id": 10,
+                "name": "slack_zulip",
+                "file_name": None,
+                "is_animated": False,
+                "deactivated": False,
+                "author": None,
+                "realm": 0,
+            },
+        ]
+        emoji_url_map = {
+            "bowtie": "https://emoji.slack-edge.com/T06NRA6HM3P/bowtie/f3ec6f2bb0.png",
+            "slack_zulip": "https://emoji.slack-edge.com/T06NRA6HM3P/slack_zulip/3d93d672c2ba4dfc.gif",
+        }
+        realm_id = 0
+
+        responses.add(
+            responses.GET,
+            re.compile(r"https://.*\.slack-edge\.com/.*\.png"),
+            body=read_test_image_file("img.png"),
+            content_type="image/png",
+            status=200,
+        )
+
+        responses.add(
+            responses.GET,
+            re.compile(r"https://.*\.slack-edge\.com/.*\.gif"),
+            body=read_test_image_file("img.png"),
+            content_type="image/gif",
+            status=200,
+        )
+
+        output_dir = tempfile.mkdtemp()
+        try:
+            for emoji_data in zerver_realmemoji:
+                emoji_name = emoji_data["name"]
+                assert isinstance(emoji_name, str)
+
+                emoji_id = emoji_data["id"]
+                emoji_url = emoji_url_map[emoji_name]
+                assert isinstance(emoji_url, str)
+                assert isinstance(emoji_id, int)
+
+                emoji_file_name = get_emojis(
+                    output_dir,
+                    emoji_url,
+                    emoji_name,
+                    emoji_id,
+                    realm_id,
+                )
+
+                assert emoji_file_name is not None
+                self.assertFalse(emoji_file_name.startswith(emoji_name))
+                self.assertRegex(emoji_file_name, r"[a-zA-Z0-9]")
+                expected_file_type = emoji_file_name.split(".")[-1]
+                self.assertTrue(emoji_file_name.endswith(expected_file_type))
```

Comment:
This assert is also meaningless -- "take everything after the last dot in this string, and assert that the string ends with that"

Given that this loop is only run twice, I think it would be far more understandable to unroll it.  Then you can explicitly and clearly assert the expected content-types and filename extensions.  And it would make meaningless asserts like this less likely.